### PR TITLE
Bug 1162048 use mozid modal

### PR DESCRIPTION
**Pull request by malena**
https://bugzilla.mozilla.org/show_bug.cgi?id=1162048

This PR replaces the old mozilla-modal styles with the new mozid-modal. This was prompted by new modal implemented on this page: https://www.mozilla.org/en-US/firefox/hello/

The following pages were updated to work with the new styles:
https://www.mozilla.org/en-US/firefox/35.0a2/firstrun/ 
http://www.mozilla.org/en-US/about/manifesto/ 
https://www.mozilla.org/en-US/firefox/geolocation/
https://www.mozilla.org/en-US/firefox/developer/ 
https://www.mozilla.org/en-US/foundation/annualreport/2012/ 
https://www.mozilla.org/en-US/foundation/annualreport/2013/
https://www.mozilla.org/en-US/firefox/new/ 
https://www.mozilla.org/en-US/firefox/os/devices/ 
https://www.mozilla.org/en-US/firefox/partners/
https://www.mozilla.org/en-US/contribute/

Modals were removed from pages no longer using them:
https://www.mozilla.org/en-US/firefox/installer-help/ - modal no longer used

This page was not updated because mozilla-modal.js was not used to implement modal
https://www.mozilla.org/en-US/foundation/annualreport/2011/

These pages are no longer accessible:
firefox_os -  template to be removed in separate bug
firefox_firstrun -  removed modal,  modal no longer used in template
firefox_os_mwc_2015_preview - updated to new modal in case this becomes active

### CODE REVIEW

**Comment by craigcook** [comment]
This needs a rebase. Sorry I let it get so stale without review.

**Comment by malena** [comment]
@craigcook No problemo! I've rebased and I have also squashed all the commits.

**Comment by alexgibson** [comment]
Close button is cropped on `about/manifesto` page:

manifesto-button

**Comment by alexgibson** [comment]
Close buttons overlaps slightly on `/firefox/os/devices/` at small viewport sizes:

firefox-os-devices

**Comment by alexgibson** [comment]
There's a red header element visible on the `/contribute` at small viewport sizes:

contribute-modal-header

File: media/css/firefox/geolocation.less [hunk 1]
```diff
@@ -30,22 +30,36 @@
...
 
-#geo-busy {
-    width: 16px;
-    height: 15px;
-    vertical-align: middle;
-}
+    #geo-demo {
+        position: relative;
+        top: 50px;
+        background: rgba(100, 100, 100, 0.5);
```
Comment (alexgibson) [inline media/css/firefox/geolocation.less]: Nit: just go for a solid color here? e.g. `#646464`

File: media/css/firefox/geolocation.less [hunk 2]
```diff
@@ -30,22 +30,36 @@
...
+        padding: 0px 0px 10px 0px;
```
Comment (alexgibson) [inline media/css/firefox/geolocation.less]: Nit: `0` values shouldn't need a `px` unit

**Comment by alexgibson** [comment]
On `/firefox/os` the newsletter form width breaks out of the modal container slightly:

screen shot 2015-07-06 at 10 31 26 am

**Comment by alexgibson** [comment]
Just a few page specific alignment things both other than that this change is looking great, thanks @malena! :+1:

File: media/css/base/mozilla-modal.less [hunk 3]
```diff
@@ -28,52 +28,63 @@ html.noscroll body {
...
-                &:hover {
-                    a {
-                        text-decoration: underline;
+                    min-width: 0;
+                    width: 42px;
+                    height: 42px;
+                    .image-replaced();
+                    border-radius: 50%;
+                    border: 2px solid #fff;
+                    .transition(all 0.2s ease);
```
Comment (alexgibson) [inline media/css/base/mozilla-modal.less]: Not related to this change, but this `transition(all 0.2s ease);` seems unneeded (not to mention heavy-handed?)

**Comment by malena** [comment]
@alexgibson I've tackled those comments. Let me know if I should squash this last commit with the first.

**Comment by alexgibson** [comment]
@malena - thanks I will take another look as soon as I can. If we're looking good, we can squash then before merging :+1:

File: media/css/firefox/os/firefox-os.less [hunk 4]
```diff
@@ -826,6 +833,7 @@ ul.apps {
     background: #fff;
     border: 0;
     box-shadow: none;
+    box-sizing: border-box;
```
Comment (alexgibson) [inline media/css/firefox/os/firefox-os.less]: Nit: use `.border-box()` mixin

Comment (malena) [inline media/css/firefox/os/firefox-os.less]: Updated to mixin.

**Comment by alexgibson** [comment]
r+ :rabbit: Thanks @malena! 

Sorry this one took a little while to land, thanks for your patience!

**Pull request merged**